streamnative/pulsar-manager
# Add support for HerdDB database

Master Issue: #177 

This patch introduces support for HerdDB as storage:
- new "init database" script
- fix references to 'timestamp' column in SQL, use MySQL escape syntax as 'timestamp' is a reserved word in HerdDB (that in turn uses Apache Calcite as SQL parser)
- fix SQL in "delete" statements to make Apache Calcite happy, as it forbids adding/subtracting unknown datatype (the '?' JDBC parameter is of type UNKNOWN for Calcite at SQL parsing time)
- fixed a query with a "GROUP BY" clause that was not grouping all of the expected columns (Apache Calcite treats that fact as an error)
- added a little example to boot HerdDB in memory-only mode (tested manually on a standalone setup)

It is possible to boot the server inside the application for persistent storage or even for replication.
In case of replication it uses ZooKeeper and BookKeeper.

Side note:
- I had to add some "Maven repositories" to make the build pass locally on my machine


In order to test this branch:
1) change the src/main/resources/application.properties file and check for comments about HerdDB
4) run pulsar manager as usual
